dashboards-observability/public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -389,13 +393,15 @@ export const Explorer = ({
       getVisualizations();
       getAvailableFields(`search source=${curIndex}`);
     } else {
-      findAutoInterval(startTime, endTime);
+      if (!selectedIntervalRef.current || selectedIntervalRef.current.text === 'Auto') {
+        findAutoInterval(startingTime, endingTime);
+      }
       if (isLiveTailOnRef.current) {
         getLiveTail(undefined, getErrorHandler('Error fetching events'));
       } else {
         getEvents(undefined, getErrorHandler('Error fetching events'));
       }
-      getCountVisualizations(minInterval);
+      getCountVisualizations(selectedIntervalRef.current!.value.length > 2 ? selectedIntervalRef.current!.value.slice(5) : selectedIntervalRef.current!.value);
```

Comment:
> `    selectedIntervalRef.current = ({ text: 'Auto', value: 'auto_' + minInterval })`

is the `auto_` needed? seems `value` can either be `auto_x` or `x` but why

Reply:
I think the main issue I can see here is that if it is just x it would be ambiguous with the other x option - i.e. if Auto is 'Auto', 'w', there may be ambiguity between auto and 'Week' 'w' options

Reply:
true. might be an opinion but can this code be more explicit? e.g. `selectedIntervalRef.current!.value.replace(/^auto_/, '')`

Reply:
Agreed and fixed in [25ed4ec](https://github.com/opensearch-project/observability/pull/1266/commits/25ed4ec02ba006512631e22aacf038f9ca599525)
[19aaff1](https://github.com/opensearch-project/observability/pull/1266/commits/19aaff1399fd9a979917986361588dd144f0ab93)